addons/point_of_sale/models/pos_order.py

```diff
@@ -144,11 +144,14 @@ def _process_order(self, order, draft, existing_order):
             pos_order._create_order_picking()
             pos_order._compute_total_cost_in_real_time()
 
-        if pos_order.to_invoice and pos_order.state == 'paid':
+        if pos_order._should_create_invoice():
             pos_order._generate_pos_order_invoice()
 
         return pos_order.id
 
+    def _should_create_invoice(self):
```

Comment:
I don't think those changes in point of sale are needed, you can just put to_invoice to 'True' in you override of 'def _order_fields(self, ui_order):' like you are overriding the partner_id